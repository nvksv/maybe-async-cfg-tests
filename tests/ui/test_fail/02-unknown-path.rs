#[maybe_async_cfg::maybe(
    sync(feature = "is_sync"), 
    unknown(not(feature = "is_sync")),
)]
async fn async_fn() -> bool {
    true
}

#[maybe_async_cfg::maybe(
    idents(async_fn(fn)),
    sync(feature = "is_sync"), 
    async(not(feature = "is_sync")),
)]
async fn test_async_fn() {
    let res = async_fn().await;
    assert_eq!(res, true);
}

fn main() {

}
