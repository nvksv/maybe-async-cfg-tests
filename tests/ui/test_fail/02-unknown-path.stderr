error[E0425]: cannot find function `async_fn_async` in this scope
  --> tests/ui/test_fail/02-unknown-path.rs:15:15
   |
15 |     let res = async_fn().await;
   |               ^^^^^^^^ not found in this scope
